frontend/src/scenes/insights/InsightRouter.tsx

```diff
@@ -0,0 +1,75 @@
+import { Card, Col, Row, Skeleton } from 'antd'
+import { kea, useValues } from 'kea'
+import { router, combineUrl } from 'kea-router'
+import api from 'lib/api'
+import { NotFound } from 'lib/components/NotFound'
+import React from 'react'
+import { DashboardItemType } from '~/types'
+import { insightRouterLogicType } from './InsightRouterType'
+
+const insightRouterLogic = kea<insightRouterLogicType<DashboardItemType>>({
+    actions: {
+        loadInsight: (id: string) => ({ id }),
+        setError: true,
+    },
+    reducers: {
+        error: [
+            false,
+            {
+                setError: () => true,
+            },
+        ],
+    },
+    listeners: ({ actions }) => ({
+        loadInsight: async ({ id }) => {
+            const response = await api.get(`api/insight/?short_id=${id}`)
+            if (response.results.length) {
+                const item = response.results[0] as DashboardItemType
+                router.actions.push(
+                    combineUrl('/insights', item.filters, {
+                        fromItem: item.id,
+                        fromItemName: item.name,
+                        fromDashboard: item.dashboard,
+                        id: item.short_id,
+                    }).url
```

Comment:
This repopulates query parameters which still isn't as immediately sharable as it could be. I thought we were working to remove reliance on the query parameters altogether so the shortId would always be present

Reply:
Ok so actually this is intended not as an object page (like `https://app.posthog.com/person/{id}`) but rather as a short link redirection (like YT, see above) because exactly as you said, until we address #4329 (which will be quite an extensive refactor), we won't be able to accomplish this.

With this current version I'm considering having a share button where we would copy this short URL to your clipboard. Wdyt?

Reply:
Tagging @liyiy because I'm not sure how this plays with the stuff you've worked on around this.

Reply:
I think this seems fine since this is a stopgap until #4239 is addressed, just wondering if we could do without a short id and just use the dashboard item's id? It didn't work when I did it because I tried to keep that id the entire time even on the insight's page but since you're using it to repopulate the query parameters, it should be alright? 😸

Reply:
Totally @liyiy! The idea here with short ID is transition to removing more reliance on the integer primary key to avoid leaking information about the number of objects in the DB. Though not sure if this is worth it in this particular case. Thoughts?

Reply:
Ohhh that's a good point 😮 I'm not sure what the implications of leaking information about the number of objects in the DB are, so I'll default to yours and others' judgement on that ^_^

Reply:
👍 think it makes sense to have less confusion around URLs, so will merge this. Thanks for the review @liyiy & @EDsCODE!